kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -604,18 +637,11 @@ private String getAssembledVendorAddress(VendorDetail vendorDetail) {
     }
 
     /**
-     * @param vendorDetail
+     * @param va
      * @return vendor country formatted for security compliance output file
      */
-    private String getVendorCountry(VendorDetail vendorDetail) {
+    private String getVendorCountry(VendorAddress va) {
         StringBuilder retval = new StringBuilder();
-        List<VendorAddress> vendorAddresses = vendorDetail.getVendorAddresses();
-        String addressType = vendorDetail.getVendorHeader().getVendorType().getAddressType().getVendorAddressTypeCode();
-        VendorAddress va = getVendorService().getVendorDefaultAddress(vendorAddresses, addressType, null);
-        if (va == null) {
-            LOG.error("No address was found for vendor " + vendorDetail.getVendorNumber());
-            return retval.toString();
-        }

```

Comment:
Moving this to #generateVendorLine() provides part of the performance enhancement.